libs/chain-signatures/contract/src/tee/tee_participant.rs

```diff
@@ -138,6 +199,52 @@ impl TeeParticipantInfo {
         }
     }
 
+    
+    fn check_app_compose_fields(tcb_info: &Value) -> bool {
+        let compose_str = match tcb_info.get("docker_compose_file").and_then(|v| v.as_str()) {
+            Some(compose) => compose,
+            None => return false,
+        };
+
+        let parsed: Config = match serde_json::from_str(compose_str) {
+            Ok(cfg) => cfg,
+            Err(_) => return false,
+        };
+
+        // Construct expected Config baseline
+        let expected = Config {
+            manifest_version: 2,
+            name: "".to_string(),  // Not validated
+            runner: "docker-compose".to_string(),
+            docker_compose_file: "".to_string(), // Not validated
+            docker_config: serde_json::json!({}),
+            kms_enabled: false,
+            gateway_enabled: false,
+            public_logs: true,
+            public_sysinfo: true,
+            local_key_provider_enabled: true,
+            allowed_envs: vec![],
+            no_instance_id: true,
+            secure_time: false,
+            pre_launch_script: "".to_string(),
+        };
+
+        // Return false on any mismatch
+        parsed.manifest_version == expected.manifest_version &&
+        parsed.runner == expected.runner &&
+        parsed.docker_config == expected.docker_config &&
+        parsed.kms_enabled == expected.kms_enabled &&
+        parsed.gateway_enabled == expected.gateway_enabled &&
+        parsed.public_logs == expected.public_logs &&
+        parsed.public_sysinfo == expected.public_sysinfo &&
+        parsed.local_key_provider_enabled == expected.local_key_provider_enabled &&
+        parsed.allowed_envs == expected.allowed_envs &&
+        parsed.no_instance_id == expected.no_instance_id &&
+        parsed.secure_time == expected.secure_time &&
+        parsed.pre_launch_script == expected.pre_launch_script
```

Comment:
Nit: Why don't we just do `parsed == expected`? (And derive PartialEq on the Config struct)
```suggestion
  parsed == expected
```